core/state_transition.go

```diff
@@ -242,6 +254,74 @@ func NewStateTransition(evm *vm.EVM, msg *Message, gp *GasPool) *StateTransition
 	}
 }
 
+func (st *StateTransition) checkGasFormula() error {
```

Comment:
Will this be part of prod code?

Reply:
I think so, if this assumption doesn't hold, something bad happens.